src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -282,6 +282,10 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     this.filterHiddenColumns();
     this.initColumnFilters();
     this.updateColumnFilterOptions();
+    // Notify all subscribers to reset their current selection.
+    if (this.updateSelection.observers.length > 0) {
```

Comment:
Any concerns removing this `if`?

Reply:
Not really, but I would suggest using `this.updateSelection?.emit(new CdTableSelection());` just to be extra safe.

Reply:
> but I would suggest using this.updateSelection?.emit(new CdTableSelection()); just to be extra safe.

Hmm, i'm not sure if this is necessary because if `this.updateSelection` is not a valid `EventEmitter` object, then something really strange has been happened and then i don't think it is necessary to check it because the only possible option is that the whole UI is badly broken then.